lib/Jsrt/Jsrt.cpp

```diff
@@ -2511,6 +2512,92 @@ CHAKRA_API JsHasException(_Out_ bool *hasException)
     return JsNoError;
 }
 
+CHAKRA_API JsGetAndClearExceptionWithMetadata(_Out_ JsValueRef *metadata)
```

Comment:
They do share a common prelude, but they do slightly different things for TTD, and I need to reference a few things from the prelude in the latter part of this function like the `scriptContext` and the `recordedException`. We could split the prelude out into a common function but it would be very specific.